src/main/java/com/trainerworkloadservice/service/TrainerInfoMongoService.java

```diff
@@ -0,0 +1,177 @@
+package com.trainerworkloadservice.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.trainerworkloadservice.dto.TrainerWorkloadServiceDto;
+import com.trainerworkloadservice.mapper.TrainerInfoMapper;
+import com.trainerworkloadservice.model.Month;
+import com.trainerworkloadservice.model.TrainerInfo;
+import com.trainerworkloadservice.model.Year;
+import com.trainerworkloadservice.repository.TrainerInfoMongoRepository;
+import com.trainerworkloadservice.utils.TrainerStatus;
+import jakarta.annotation.PostConstruct;
+import jakarta.jms.JMSException;
+import jakarta.jms.TextMessage;
+import java.util.HashMap;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.activemq.Message;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+@Data
+public class TrainerInfoMongoService {
+
+  private final TrainerInfoMongoRepository repository;
+  private final ObjectMapper objectMapper;
+  private final TrainerInfoMapper trainerInfoMapper;
+
+  @PostConstruct
+  public void init() {
+    initializeTestData();
+  }
+
+  public TrainerInfo findByUsername(String username) {
+    return repository.findByUsername(username);
+  }
+
+  public TrainerInfo saveTrainer(TrainerInfo trainer) {
+    return repository.save(trainer);
+  }
+
+  public void initializeTestData() {
+    String testUsername = "test_trainer";
+
+    TrainerInfo existingTrainer = repository.findByUsername(testUsername);
+
+    if (existingTrainer == null) {
+      TrainerInfo trainer = new TrainerInfo();
+      trainer.setUsername(testUsername);
+      trainer.setFirstName("John");
+      trainer.setLastName("Doe");
+      trainer.setStatus(TrainerStatus.ACTIVE);
+
+      Year year2024 = new Year(2024);
+      Month month = new Month(new HashMap<>());
+      month.getMonthDurations().put("november", 10);
+      year2024.getMonths().add(month);
+
+      trainer.getYears().add(year2024);
+
+      repository.save(trainer);
+
+      log.info("Test data has been initialized!");
+    } else {
+      log.info("Test data already exists. Skipping initialization.");
+    }
+  }
+
+  @JmsListener(destination = "trainer.workload.queue")
```

Comment:
You have the same listener in TrainerInfoService. Which one will work?

Reply:
both work in parallel